src/devices/Tcs3472x/Tcs3472x.cs

```diff
@@ -0,0 +1,227 @@
+        }
+
+        private void PowerOn()
+        {
+
+            _i2cDevice.Write(new byte[] { (byte)(Registers.COMMAND_BIT | Registers.ENABLE), (byte)Registers.ENABLE_PON });
+            Thread.Sleep(10);
+            _i2cDevice.Write(new byte[] { (byte)(Registers.COMMAND_BIT | Registers.ENABLE), (byte)(Registers.ENABLE_PON | Registers.ENABLE_AEN) });
+        }
+
+        private void PowerOff()
+        {
+            var powerState = I2cRead8(Registers.ENABLE);
+            powerState = (byte)(powerState & ~(byte)(Registers.ENABLE_PON | Registers.ENABLE_AEN));
+            _i2cDevice.Write(new byte[] { (byte)(Registers.COMMAND_BIT | Registers.ENABLE), powerState });
+        }
+
+        /// <summary>
+        /// Set/Clear the colors and clear interrupts
+        /// </summary>
+        /// <param name="state">true to set all interrupts, false to clear</param>
+        public void SetInterrupt(bool state)
+        {
+            SetInterrupt(InterruptState.All, state);
+        }
+
+        /// <summary>
+        /// Set/clear a specific interrupt persistence
+        /// </summary>
+        /// <param name="interupt">The percistence cycles</param>
+        /// <param name="state">True to set the interrupt, false to clear</param>
+        public void SetInterrupt(InterruptState interupt, bool state)
+        {
+            _i2cDevice.Write(new byte[] { (byte)(Registers.COMMAND_BIT | Registers.PERS), (byte)interupt });
+            var enable = I2cRead8(Registers.ENABLE);
+            enable = state ? enable |= (byte)Registers.ENABLE_AIEN : enable = (byte)(enable & ~(byte)Registers.ENABLE_AIEN);
+            _i2cDevice.Write(new byte[] { (byte)(Registers.COMMAND_BIT | Registers.ENABLE), enable });
+        }
+
+        /// <summary>
+        /// Get the color
+        /// </summary>
+        /// <param name="delay">Wait to read the data that the integration time is passed</param>
+        /// <returns></returns>
+        public Color GetColor(bool delay = true)
+        {
+
+            if (delay)
+                Thread.Sleep((int)(IntegrationTime * 1000));
+            var divide = ((256 - _integrationTimeByte) * 1024);
+            // If we are in long wait, we'll need to divide even more
+            if (_isLongTime)
+                divide *= 12;
+            int r = (int)(I2cRead16(Registers.RDATAL) * 255 / divide);
+            r = Math.Clamp(r, 0, 255);
+            int g = (int)(I2cRead16(Registers.GDATAL) * 255 / divide);
+            g = Math.Clamp(g, 0, 255);
+            int b = (int)(I2cRead16(Registers.BDATAL) * 255 / divide);
+            b = Math.Clamp(b, 0, 255);
+            int a = (int)(I2cRead16(Registers.CDATAL) * 255 / divide);
+            a = Math.Clamp(a, 0, 255);
+            return Color.FromArgb(a, r, g, b);
+        }
+
+        /// <summary>
+        /// Get the color
+        /// </summary>
+        public Color Color => GetColor();
```

Comment:
GetColor should be internal/private

Reply:
I leave it public if you want to read the colors even if integration cycle is not over. So read the previous value for example